### ISSUE TITLE
Feature request: Typewriter and Focus modes

### ISSUE THREAD

**Issue by 0x63616a7573** [notaapp/issues#38]
Same options/functionality as previously available in Caret under the View menu:

- Typewriter mode: where the line currently being edited is always vertically centered
- Focus mode: where text _other_ than the line/text block currently being edited is dimmed

**Comment by astoilkov** [comment]
Thanks for the vote. We will consider both features again.

We aren't sure about both features yet because previously they were creating maintenance problems. We should see if the benefits outweigh the disadvantages.

**Comment by manuraj17** [comment]
Voting again for typewriter mode. Everything else is as I wish to be, except this mode. I happen to write long notes and it really helps if the note is centered to screen. At times I end up writing on a laptop so, having it at the center helps.

**Comment by astoilkov** [comment]
Thanks @manuraj17 

This thread talks about two different features. For others voting in this thread, it may be useful if you say which of the two you are interested in.

**Comment by manuraj17** [comment]
@astoilkov Oh okay. I am voting for the **typewriter mode** 😄  (I had mentioned it)

> Typewriter mode: where the line currently being edited is always vertically centered

Although another approach is also fine, even if the current line is not strictly centered all the time -- the ability to scroll up and thereby moving the last line to end the middle of the page; Which results in having some extra space when we are nearing the page end also works.

**Comment by astoilkov** [comment]
> the ability to scroll up and thereby moving the last line to end the middle of the page; Which results in having some extra space when we are nearing the page end also works.

Yes. That makes sense. We also don't like when we have to write at the end of a document and can't put that line in the center or close to the center.

**Comment by manuraj17** [comment]
@astoilkov Exactly. There needn't be any strict white space in the document or always be centered --  it's just ability to scroll up a bit when we are at the last line so that it comes to the center, just like you said. The difference that would make is huge. Hope to see this come in soon! (As of now in long notes I have to move out of nota and use other apps -_- )

PS: I see that I have made some gramatical mistakes in the above statement, but I believe you got the idea :D 

Thanks!

**Comment by Auratelience** [comment]
Voting for both typewriter and focus mode.

**Comment by fedorovpishet** [comment]
Add my voice for typewriter and focus mode. I'm a writer and it's very useful features. I use to used it on iA Writer, Obsidian and Ulysses, but now I [move to Nota](https://t.me/fedorovpishet2/191). Hope you'll add this features. 

P.S. Anyway you're awesome)

**Comment by johnjago** [comment]
This makes a big difference for people who write long pieces on small screens. I usually write blog posts on a laptop screen, so I end up looking at the bottom of the screen where the text is. As a workaround, I hit enter a few times to push the text up, but it would be so nice to have this built-in 🙂
